### PR TITLE
ILS: case 223423

### PR DESCRIPTION
**Pull request by kkrampa**
@gcapalbo 
Mike has recently discovered a problem in reports: http://manage.dimagi.com/default.asp?223423.
I've realized that some non-facility locations were recalculated in function `process_facility_data` which caused problems in data. It affects only data from February and earlier data.  

My plan is to first recalculate one location and check if it fixes reporting data. If so, I will run recalculation for rest of locations.

### CODE REVIEW

File: custom/ilsgateway/oneoff.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+        ))
+        last_processed = non_facilities[0]
+        non_facilities = non_facilities[1:]
+        process_non_facility_warehouse_data(last_processed.couch_location, start_date, end_date, strict=False)
+
+    for non_facility in non_facilities:
+        task_progress.last_synced_object_id = non_facility.location_id
+        task_progress.progress += 1
+        task_progress.save()
+        process_non_facility_warehouse_data(non_facility.couch_location, default_start_date(), end_date, strict=False)
```
Comment (dimagimon) [inline custom/ilsgateway/oneoff.py]: E501 line too long (118 > 115 characters)

File: custom/ilsgateway/models.py [hunk 2]
```diff
@@ -550,7 +561,7 @@ def location_edited_receiver(sender, loc, moved, **kwargs):
         )
 
     group = last_location_group(loc)
-    if group != loc.metadata['group']:
+    if not loc.sql_location.location_type.administrative and group != loc.metadata['group']:
```
Comment (kkrampa) [inline custom/ilsgateway/models.py]: There is possibility to set up group for non-facility location in application but it doesn't make sense from reports perspective so we can just ignore that.

File: custom/ilsgateway/models.py [hunk 3]
```diff
@@ -485,6 +485,17 @@ class Meta:
         app_label = 'ilsgateway'
 
 
+class OneOffTaskProgress(models.Model):
```
Comment (gcapalbo) [inline custom/ilsgateway/models.py]: i realize this is a custom model but i think it's still important to put domain on here and make sure that we filter on domain everywhere it's used

**Comment by gcapalbo** [comment]
@kkrampa just one comment otherwise looks good

**Comment by kkrampa** [comment]
@gcapalbo Updated

**Pull request merged**